### PR TITLE
Continue #1081 (PubSub)

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

File: test/cli/pubsub.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const createTempNode = ''
 const repoPath = require('./index').repoPath
```
Comment (diasdavid) [inline test/cli/pubsub.js]: @richardschneider are these tests supposed to be enabled?

Comment (diasdavid) [reply test/cli/pubsub.js]: image

**Comment by diasdavid** [comment]
Still 3 errors:

```
  1) cli
       pubsub
         "before all" hook:
     TypeError: createTempNode is not a function
      at Context.before (test/cli/pubsub.js:25:5)

  2) cli
       pubsub
         "after all" hook:
     TypeError: Cannot read property 'goOffline' of undefined
      at Context.after (test/cli/pubsub.js:33:10)

  3) HTTP API
       ## http-api spec tests
         /pubsub
           /pub
             returns 500 if no buffer is provided:
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

The first two come from an incomplete test (`createTempNode = ''`) and the third one is a bug.

File: test/cli/pubsub.js (hunk 1 above)
Comment (richardschneider) [reply test/cli/pubsub.js]: I did comment somewhere, that the test is very old and doesn't really work,  By doing `require('./index').repoPath`, it brings in all the other tests.

**Comment by diasdavid** [comment]
@pgte could you give a hand on this one?

**Comment by pgte** [comment]
Sure

**Comment by pgte** [comment]
@diasdavid @richardschneider CI is green, I think I'm done here.

**Comment by diasdavid** [comment]
Phenomenal @pgte \o/ thank you. Will just wait for travis to finish for the merge :)

**Comment by pgte** [comment]
Spoke too soon…
@diasdavid any idea why CI failed? https://travis-ci.org/ipfs/js-ipfs/jobs/305843248#L3850

**Comment by diasdavid** [comment]
@pgte given that all the errors come from from tests that touch the http-api, something tells me that the changes in js-ipfs-api made it more wonky when run on Travis. I also noticed that hapi had a recent update. It also seems that most of the failures are just cascading failures after a timeout, perhaps increase those to fix it?

https://travis-ci.org/ipfs/js-ipfs/jobs/305843344#L5343-L5433

**Comment by pgte** [comment]
@diasdavid may be, but I find that difficult, as, for instance,`bitswap.stat` timed out, and we didn't touch that one recently..
Anyway, I'll be increasing the timeouts for the tests on the errors observed to see if it is solved..

**Comment by diasdavid** [comment]
Seems that @richardschneider is making good progress, only 3 tests remaining https://travis-ci.org/ipfs/js-ipfs/jobs/305961912#L3964-L3976

**Comment by diasdavid** [comment]
image

YEAAAAS!

**Comment by diasdavid** [comment]
Going ahead and merging this one!

**Pull request merged**